cmake/CPM.cmake

```diff
@@ -354,6 +354,33 @@ function(cpm_parse_add_package_single_arg arg outArgs)
   )
 endfunction()
 
+# Check that the working directory for a git repo is clean
+function(cpm_check_working_dir_is_clean repoPath isClean)
+  # Not sure this check is necessary given git is probably key to the rest of the script
+  find_package(Git REQUIRED)
+
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} status --porcelain
+    RESULT_VARIABLE result
+    OUTPUT_VARIABLE status
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_QUIET
+    WORKING_DIRECTORY ${repoPath}
+  )
```

Comment:
I beliebe this can create false positives, e.g. if the dependency is non-git but stored inside an outer git repository. Perhaps we should only run the check if the repo itself is really a git directory, e.g. we could check if `${repoPath}/.git` exists.

Reply:
to work around i added git rev-parse --absolute-git-dir to get the location of .git, to compare it with ${repoPath}